tests/openwisp2/urls.py

```diff
@@ -1,10 +1,32 @@
+import os
+
 from django.conf.urls import include, url
 from django.contrib import admin
 from django.contrib.staticfiles.urls import staticfiles_urlpatterns
 
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
     url(r'^accounts/', include('allauth.urls')),
+    url(r'^api/v1/', include('openwisp_utils.api.urls')),
 ]
 
 urlpatterns += staticfiles_urlpatterns()
+
+if os.environ.get('SAMPLE_APP', False):
+    # Load url configuration for custom api views: This should be done when
+    # you are extending the app and modifing the API views in your extended
+    #  application.
+    urlpatterns += [
+        url(
+            r'^',
+            include(
+                'openwisp2.sample_notifications.urls', namespace='sample_notifications'
+            ),
+        )
+    ]
+else:
+    # Load openwisp_notifications api views: This can be used when you are
+    # extending the app but not making any changes in the API views.
+    urlpatterns += [
+        url(r'', include('openwisp_notifications.urls'),),
+    ]
```

Comment:
what about:

```python
notifications_app = 'openwisp_notifications' if not os.environ.get('SAMPLE_APP', False) else 'sample_notifications.urls'

urlpatterns += [
    url(r'', include(f'{notifications_app}.urls'),),
]
```

Is `namespace='sample_notifications'` required? If not, we can leave it out.

Reply:
I used `namespace` to avoid conflicts with URL names of other applications (by end-user). I don't think it will be of best interest to write the logic using ternary expressions since we ask users to refer to this file in README. Maybe being verbose will help them better understand the concept.